operator/api/v1alpha1/observatorium_types.go

```diff
@@ -97,11 +98,36 @@ type StoreCacheSpec struct {
 	MemoryLimitMB *int32 `json:"memoryLimitMb,omitempty"`
 }
 
+// APIRBAC represents a set of Observatorium API RBAC roles and role bindings.
+type APIRBAC struct {
+	// Roles is a slice of Observatorium API roles.
+	Roles []rbac.Role `json:"roles"`
+	// RoleBindings is a slice of Observatorium API role bindings.
+	RoleBindings []rbac.RoleBinding `json:"roleBindings"`
+}
+
+// APITenant represents a tenant in the Observatorium API.
+type APITenant struct {
+	Name string `json:"name"`
+	ID   string `json:"id"`
+	OIDC struct {
+		ClientID      string `json:"clientID"`
+		ClientSecret  string `json:"clientSecret"`
+		IssuerURL     string `json:"issuerURL"`
+		RedirectURL   string `json:"redirectURL"`
+		UsernameClaim string `json:"usernameClaim"`
+	} `json:"oidc"`
```

Comment:
@squat Can we switch to a named struct here. I hit this issue with controller-gen:
https://github.com/kubernetes-sigs/controller-tools/pull/415